### PR TITLE
develop-improvements

### PR DESCRIPTION
**Pull request by diegoveloper**
working on issue #216

### CODE REVIEW

File: app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 */
 package com.belatrixsf.connect.networking.retrofit.api;
 
+import com.belatrixsf.connect.entities.Category;
 import com.belatrixsf.connect.entities.Employee;
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java]: mmmm

File: app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java [hunk 2]
```diff
@@ -37,6 +38,7 @@
 
 /**
  * Created by PedroCarrillo on 4/26/16.
+ * modified by dvelasquez
  */
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java]: mmmmmmmmmmm

File: app/src/main/java/com/belatrixsf/connect/services/contracts/StarService.java [hunk 3]
```diff
@@ -20,6 +20,7 @@
 */
 package com.belatrixsf.connect.services.contracts;
 
+import com.belatrixsf.connect.entities.Category;
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/services/contracts/StarService.java]: i don't see this class being used

File: app/src/main/java/com/belatrixsf/connect/ui/account/AccountFragment.java [hunk 4]
```diff
@@ -231,19 +266,14 @@ public void onSaveInstanceState(Bundle outState) {
     }
 
     private void restoreState(Bundle savedInstanceState) {
-        List<SubCategory> subCategoryList = savedInstanceState.getParcelableArrayList(SUB_CATEGORY_LIST_KEY);
         Employee employee = savedInstanceState.getParcelable(EMPLOYEE_KEY);
         byte [] employeeImg = savedInstanceState.getByteArray(AccountActivity.USER_IMG_PROFILE_KEY);
-        accountPresenter.loadPresenterState(subCategoryList, employee, employeeImg);
+        accountPresenter.loadPresenterState(employee, employeeImg);
     }
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/ui/account/AccountFragment.java:251]: not saving the subcategories anymore?

File: app/src/main/java/com/belatrixsf/connect/ui/account/AccountFragment.java [hunk 5]
```diff
@@ -434,12 +458,12 @@ public void hideProgressIndicator() {
 
     @Override
     public void showNoDataView() {
-        noDataTextView.setVisibility(View.VISIBLE);
+        //noDataTextView.setVisibility(View.VISIBLE);
     }
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/ui/account/AccountFragment.java]: will always be populated? 😮

File: app/src/main/java/com/belatrixsf/connect/ui/account/AccountPresenter.java [hunk 6]
```diff
@@ -87,11 +86,11 @@ public void onFailure(ServiceError serviceError) {
                 employeeService.getEmployee(employeeId, employeeBelatrixConnectCallback);
             }
         } else {
-            loadSubCategoriesStar(false);
             showEmployeeData();
         }
     }
 
+    /*
     private void loadSubCategoriesStar(boolean force) {
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/ui/account/AccountPresenter.java]: oh ok, no more subcategories

File: app/src/main/java/com/belatrixsf/connect/ui/recommendations/RecommendationsFragment.java [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        saveState(outState);
+        super.onSaveInstanceState(outState);
+    }
+
+    private void restoreState(Bundle savedInstanceState) {
+        List<Star> list = savedInstanceState.getParcelableArrayList(_RECOMMENDATION_LIST_KEY);
+        recommendationsPresenter.loadPresenterState(list);
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/ui/recommendations/RecommendationsFragment.java:168]: i don't see it being saved

**Review by PedroCarrillo — COMMENTED** [review]
Changes are good just a lot of empty spaces found in new classes and unused imports

File: app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java (hunk 2 above)
Comment (diegoveloper) [reply app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java]: :)

**Pull request merged**